src/pages/workspace/WorkspaceNewRoomPage.js

```diff
@@ -76,19 +82,59 @@ class WorkspaceNewRoomPage extends React.Component {
         this.setState({workspaceOptions: _.map(workspaces, policy => ({label: policy.name, key: policy.id, value: policy.id}))});
     }
 
+    validateAndCreatePolicyRoom() {
+        if (!this.validate()) {
+            return;
+        }
+        Report.createPolicyRoom(
+            this.state.policyID,
+            this.state.roomName,
+            this.state.visibility,
+        );
+    }
+
     /**
-     * Called when a workspace is selected.
-     * @param {String} policyID
+     * @returns {Boolean}
      */
-    onWorkspaceSelect(policyID) {
-        this.setState({policyID});
+    validate() {
+        const errors = {};
+
+        // We error if the user doesn't enter a room name or left blank
+        if (!this.state.roomName || this.state.roomName === CONST.POLICY.ROOM_PREFIX) {
+            errors.roomName = this.props.translate('newRoomPage.pleaseEnterRoomName');
+        }
+
+        // We error if the room name already exists. We don't care if it matches the original name provided in this
+        // component because then we are not changing the room's name.
```

Comment:
I don't really understand the comment. In particular...

>We don't care if it matches the original name provided in this component because then we are not changing the room's name.

What does this have to do with the check we make? How is it determined that the name in state matches the "original name provided in this component"? Which variable represents that "original name"? Can we be more specific or is the comment out of context?

Reply:
@marcaaron Yes actually it is out of context for this page so we should remove this and we can be more specific on same comment on `ReportSettingsPage` by just modifing it as `We don't error if the room name matches same as previous.`